### PR TITLE
Set default value for nccl make MAX_JOBS if ProcessorCount returns 0

### PR DESCRIPTION
**Pull request by mrshenli**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #84231
* #84221

### CODE REVIEW

**Comment by mrshenli** [comment]
The error message I got is:

```
[108/6838] Performing build step for 'nccl_external'                                                                                                              
FAILED: nccl_external-prefix/src/nccl_external-stamp/nccl_external-build nccl/lib/libnccl_static.a /fsx/users/shenli/pytorch/build/nccl_external-prefix/src/nccl_e
xternal-stamp/nccl_external-build /fsx/users/shenli/pytorch/build/nccl/lib/libnccl_static.a                                                                       
cd /fsx/users/shenli/pytorch/third_party/nccl/nccl && env CCACHE_DISABLE=1 SCCACHE_DISABLE=1 make -j0 -l0 CXX=/data/shared/bin/c++ CUDA_HOME=/usr/local/cuda-11.1
NVCC=/data/shared/bin/cuda/nvcc NVCC_GENCODE=-gencode=arch=compute_80,code=sm_80 BUILDDIR=/fsx/users/shenli/pytorch/build/nccl VERBOSE=0 && /fsx/users/shenli/cond
a/envs/compiler/bin/cmake -E touch /fsx/users/shenli/pytorch/build/nccl_external-prefix/src/nccl_external-stamp/nccl_external-build
make: the '-j' option requires a positive integer argument
```

File: cmake/External/nccl.cmake [hunk 1]
```diff
@@ -38,6 +42,7 @@ if(NOT __NCCL_INCLUDED)
       BUILD_IN_SOURCE 1
       CONFIGURE_COMMAND ""
       BUILD_COMMAND
+        env
```
Comment (malfet) [inline cmake/External/nccl.cmake:45]: Why this is needed?

**Review by malfet — APPROVED** [review]
LGTM if CI is green (why `env` is needed though)?

File: cmake/External/nccl.cmake (hunk 1 above)
Comment (rohan-varma) [inline cmake/External/nccl.cmake:45]: is this change intentional?

**Review by rohan-varma — APPROVED** [review]

File: cmake/External/nccl.cmake (hunk 1 above)
Comment (mrshenli) [reply cmake/External/nccl.cmake:45]: It was there before, but removed by #83696. I thought we need to pass this to pass env vars to make if necessary?

Comment (mrshenli) [reply cmake/External/nccl.cmake:45]: Yep, it was there before, but removed by https://github.com/pytorch/pytorch/pull/83696. I thought we need to pass this to pass env vars to `make` if necessary?

Comment (peterbell10) [reply cmake/External/nccl.cmake:45]: No `env` vars are being passed though, it was left over from the `CCACHE_DISABLE=1` that was there before.

Comment (mrshenli) [reply cmake/External/nccl.cmake:45]: Oh I see, let me remove that then.

**Comment by mrshenli** [comment]
@@pytorchbot merge -g
